### PR TITLE
[Droid] Creating an extension method around CoreDispatcher.RunAsync leads to cast exception

### PR DESCRIPTION
**Pull request by crogeniks**
GitHub Issue (If applicable): #

Fixes https://github.com/unoplatform/uno/issues/2044

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Creating an Extension around CoreDispatcher.RunAsync leads to a cast exception between UIAsyncAction et AsyncAction

## What is the new behavior?

Type check the IAsyncAction before casting

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Core/CoreDispatcher/CoreDispatcher_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using SamplesApp.UITests.TestFramework;
+using Uno.UITest.Helpers;
+using Uno.UITest.Helpers.Queries;
+
+namespace SamplesApp.UITests.Windows_UI_Core.CoreDispatcher
+{
+  [TestFixture]
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Core/CoreDispatcher/CoreDispatcher_Tests.cs]: This test should be a runtime test (UI Tests are expensive to run). Could you transfer it there? 

https://github.com/unoplatform/uno/tree/master/src/Uno.UI.RuntimeTests/Tests

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Core/CoreDispatcher/CoreDispatcher_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	public void RunAsyncExtensions()
+	{
+	  Run("UITests.Shared.Windows_UI_Core.CoreDispatcherControl.CoreDispatcherControl");
+
+	  _app.Marked("RunAsyncButton").Tap();
+
+
+	  Thread.Sleep(1200);
+
+	  _app.Marked("Result").WithText("Success");
```
Comment (carldebilly) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Core/CoreDispatcher/CoreDispatcher_Tests.cs]: You can replace this pattern by
```csharp
_app.WaitForDependencyPropertyValue(_app.Marked("Result"), "Text", "Success");
```

So you won't need to introduce an abritary waiting delay...

Comment (crogeniks) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Core/CoreDispatcher/CoreDispatcher_Tests.cs]: As per Jerome's request, I've moved my test to the Runtime tests section and removed this UI test. But thanks for the tip!

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Core/Given_CoreDispatcher.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Foundation;
+using Windows.System;
+using Windows.UI.Core;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Core
+{
+  #if __ANDROID__ || __IOS__
+  [TestClass]
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Core/Given_CoreDispatcher.cs]: Thanks for the update! I'm curious though, the indendation is not correct, but it should have been fixed by the `.editorconfig` file. Do you know if its activated properly ?

**Pull request merged**